### PR TITLE
Hoist the Tracking package into Connection.

### PR DESCRIPTION
**Pull request by zinigor**
This change continues the hoisting project and hoists the Tracking package into Connection. No functional changes in this PR.

#### Changes proposed in this Pull Request:

* Moves code to the Connection package.
* Moves test suites to the Connection package.
* Removes the Tracking dependency from every composer.json file.
* Marks the Tracking page as abandoned. 

#### Jetpack product discussion

* p9dueE-4Ep-p2

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Test the Tracking code using its PHPUnit test suite.
* Test Tracking by going to the dashboard. The client side part of it is easy to test using the debug switch:
* Enter this totally safe code into your JavaScript console: `localStorage.setItem('debug', 'dops:analytics');`
* This will turn on tracking debugging, you should see events being logged. Make sure you have tracking protection disabled in your browser.

### CODE REVIEW

File: projects/packages/connection/legacy/class-jetpack-tracks-client.php [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+		// Your application-specific properties
+		'custom_property'    => $some_value,
+	) );
+
+	if ( is_wp_error( $result ) ) {
+		// Handle the error in your app
+	}
+```
+ */
+class Jetpack_Tracks_Client {
```
Comment (kraftbj) [inline projects/packages/connection/legacy/class-jetpack-tracks-client.php:48]: @zinigor I haven't tested this, but do we need some type of checking to ensure there isn't also another plugin with an older version of Tracking already?

Thinking outloud, I'm running Jetpack latest with this new Connection package (which would load over older ones), but also running ... Backup plugin which requires an older Connection package and the Tracking package, how do we ensure these two both don't load?

Comment (zinigor) [reply projects/packages/connection/legacy/class-jetpack-tracks-client.php:48]: That's a great point, thank you! I'll add testing instructions for that case tomorrow, but looking at our usage of these files it looks like we're not using them outside the autoloader functionality anymore. This should make sure that we're safe. Do you think we should add a class check anyway?

**Comment by sdixon194** [comment]
Looks like this has some branch conflicts that need resolving and I think one of the testing instructions is still TBD?

File: projects/packages/connection/legacy/class-jetpack-tracks-client.php (hunk 1 above)
Comment (zinigor) [reply projects/packages/connection/legacy/class-jetpack-tracks-client.php:48]: So I have tested this finally with the latest Autoloader helper, and it seems that we have had this working before in the same way:
image

You can see here that the class has been there in the top namespace in other plugins too, including the latest Jetpack products in the wild currently. So we're not changing anything, just moving the file.

**Comment by zinigor** [comment]
I have added testing instructions, but I feel like this is growing too much because of master merges. Do you think a rebase would help?

**Review by sergeymitr — DISMISSED** [review]
👍

**Review by sergeymitr — APPROVED** [review]
👍 👍

**Pull request merged**